homeassistant/components/raspihats.py

```diff
@@ -0,0 +1,251 @@
+"""
+Support for controlling raspihats boards.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/raspihats/
+"""
+import logging
+import threading
+import time
+
+from homeassistant.const import (
+    EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP
+)
+
+REQUIREMENTS = ['raspihats==2.2.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'raspihats'
+
+CONF_I2C_HATS = 'i2c_hats'
+CONF_BOARD = 'board'
+CONF_ADDRESS = 'address'
+CONF_CHANNELS = 'channels'
+CONF_INDEX = 'index'
+CONF_INVERT_LOGIC = 'invert_logic'
+CONF_INITIAL_STATE = 'initial_state'
+
+I2C_HAT_NAMES = [
+    'Di16', 'Rly10', 'Di6Rly6',
+    'DI16ac', 'DQ10rly', 'DQ16oc', 'DI6acDQ6rly'
+]
+
+I2C_HATS_MANAGER = None
+
+
+# pylint: disable=unused-argument
+def setup(hass, config):
+    """Setup the raspihats component."""
+    # pylint: disable=global-statement
+    global I2C_HATS_MANAGER
```

Comment:
Please don't use `global`. Instead store your manager in `hass.data`, which is a dictionary for exactly this purpose.